README.md

```diff
@@ -1,31 +1,45 @@
-# TODO-MVP-ContentProviders
+# todo-mvp-contentproviders
 
-It is based on the [TODO-MVP-Loaders](https://github.com/googlesamples/android-architecture/tree/todo-mvp-loaders) sample and uses a Content Provider to retrieve data into the repository.
+This version of the app is called todo-mvp-contentproviders. It is based on the [todo-mvp-loaders](https://github.com/googlesamples/android-architecture/tree/todo-mvp-loaders) sample, and adds a [content provider]((https://developer.android.com/guide/topics/providers/content-providers.html)) to manage access to the underlying SQLite database storage. Content providers provide a standard interface for allowing access to data in one process from code running in a different process. This sample demonstrates how to add a content provider to the [todo-mvp-loaders](https://github.com/googlesamples/android-architecture/tree/todo-mvp-loaders) sample. 
 
-<img src="https://github.com/googlesamples/android-architecture/wiki/images/mvp-contentproviders.png" alt="Diagram"/>
+You could then use content providers to support additional features that are not covered by this sample, providing the following possible benefits:
 
-The advantages of Content Providers, from the [Content Provider documentation page](http://developer.android.com/guide/topics/providers/content-providers.html), are:
+ * Allow you to securely share data stored in your application with other applications.
+ * Add support for custom searches in your application.
+ * Develop widgets to access data in your application.
+ * Provide support for copying and pasting data between your application and other applications.
 
-  * Manage access to a structured set of data
-  * Content providers are the standard interface that connects data in one process with code running in another process
+Content providers can also be used to manage access to other types of storage including non-relational databases, or data stored on a file system. You can also use the [`FileProvider`](https://developer.android.com/reference/android/support/v4/content/FileProvider.html) subclass of [`ContentProvider`](https://developer.android.com/reference/android/content/ContentProvider.html) to facilitate secure sharing of files associated with an app. A [`FileProvider`](https://developer.android.com/reference/android/support/v4/content/FileProvider.html) can be used to create a `content://` URI for a file instead of a `file:///` URI. As with other versions of the [todo-mvp](https://github.com/googlesamples/android-architecture/tree/todo-mvp) app, this sample uses a SQLite database to provide data storage.
```

Comment:
This addresses a concern you and I have both expressed that content providers are often misunderstood to be directly related to SQL storage. We had agreed that the FileProvider example illustrated that other types of storage are supported so I'm leaving this as is.